services/rsources/http/http.go

```diff
@@ -98,12 +99,20 @@ func (h *handler) failedRecords(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "job_run_id not found", http.StatusBadRequest)
 		return
 	}
+	var paging rsources.PagingInfo
+	if pageSize, ok := r.URL.Query()["pageSize"]; ok && len(pageSize) > 0 {
```

Comment:
maybe we can have a default limit here(10k records)
And also an upper limit, if the user of the API sends a very large limit, it could affect the system adversely.

Reply:
The API is backwards compatible, i.e. no default limit should be applied here.
The advised page limit for clients is be really high, e.g. 500K-1Mil, thus an upper limit isn't critical imo

Reply:
>backwards compatible

Touché
But in case they use the newer API, wouldn't we want some configurable limit?

>page limit for clients is be really high

I didn't fully understand this, please elaborate

Reply:
1. Performance will suffer if page size is smaller than 500k (see performance test)
2. The payload is not significant since the endpoint practically returns a list of record ids, thus it should be fine to use pretty large page sizes